src/ReadingBuddy.java

```diff
@@ -804,40 +853,49 @@ public static void chooseFromAllBooks(){
             System.out.print("\nEnter the corresponding number to select a book to find a book buddy (make a Book Buddy" +
              "Wanted post or answer someone else's), or 0 to return to the menu:");
             menuSelection = validateMenuSelection(0, allBooks.size());
+            //TESTING
+            System.out.println("!menuSelection = " + menuSelection + "\n");
+
             if(menuSelection > 0){
                 //!!!!!MOVE THE CHECKING OF IF OPEN SEARCH TO BEFORE
+                //TESTING
+                System.out.println("in menuSelection > 0 if");
+
                 selectedBook = allBooks.get(menuSelection-1);
-                posts = searchForSearchPosts(selectedBook.getTitle(), selectedBook.getAuthor());
-                if(posts.size() > 0){
-                    if(currentUser.checkIfInCurrentBooks(selectedBook.getID()) >= 0){
-                        currentSelectedBook = currentUser.getSpecificCurrentBook(selectedBook.getID());
-                        System.out.println("\n" + (currentSelectedBook.getGroup() > 0) + "\n");
-                        if(currentSelectedBook.getHasOpenSearch()){
-                            System.out.println("\nLooks like you already have a post for that book!\n");
-                            followUp = true;
-                        }else if(currentSelectedBook.getGroup() > 0){
-                            System.out.println("\nYou are already in a book group for that book!\n");
-                            followUp = true;
-                        }else{
-                            seePosts = confirmSelection("Would you like to see other users' Book Buddy Wanted posts for this book?");
-                            if(seePosts == 1){
-                                menu2Selection = displayAndGetSearchPostSelection(posts);
-                                if (menu2Selection > 0 && menu2Selection <= posts.size()){
-                                    createNewBookGroup(posts.get(menuSelection-1));
-                                    stayInMenu = false;
-                                }
```

Comment:
altered the structure of the function to catch, if the user already has the book in their currently reading, if the user already has a wanted post or a book group for that book (and only offer the option to proceed when the user is eligible to do so).